Userland/Libraries/LibC/shadow.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
```

Comment:
I don't remember writing this code. :)

Reply:
Well, technically you did, sort of. It's mostly a copy of `pwd.cpp`. I guess I'll add myself there, then. :)